pyfolio/perf_attrib.py

```diff
@@ -114,14 +136,24 @@ def perf_attrib(returns, positions, factor_returns, factor_loadings,
 
     if len(missing_stocks) > 0:
 
-        warnings.warn("Could not find factor loadings for the following "
-                      "stocks: {}. Ignoring for exposure calculation and "
-                      "performance attribution. Coverage ratio: {}/{}. "
-                      "Average allocation of missing stocks: {} "
-                      .format(list(missing_stocks),
-                              num_stocks_covered,
-                              num_stocks,
-                              positions[missing_stocks].mean()))
+        missing_stocks_warning_msg = (
+            "Could not determine risk exposures for some of this algorithm's "
+            "positions. PnL from the missing assets will not be properly "
+            "accounted for in performance attribution.\n"
+            "\n"
+            "The following assets were missing factor loadings: {}. "
+            "Ignoring for exposure calculation and performance attribution. "
+            "Ratio of assets missing: {}. Average allocation of missing "
+            "assets:\n"
+            "\n"
+            "{}.\n"
+        ).format(
+            list(missing_stocks),
```

Comment:
should we also truncate this list if it's too long?

Reply:
Planning to do this in a follow-up PR

Reply:
Same with missing dates, don't want to list all of them but just the period.

Reply:
Yup - @twiecki that's another issue that we're tracking